spec/features/idv/steps/in_person/state_id_step_spec.rb

```diff
@@ -8,28 +8,244 @@
-    expect(page).to have_field(t('in_person_proofing.form.state_id.zipcode'), with: '')
-    # enter valid characters, but invalid length
-    fill_in t('in_person_proofing.form.state_id.zipcode'), with: '123'
-    click_idv_continue
-    expect(page).to have_css('.usa-error-message', text: t('idv.errors.pattern_mismatch.zipcode'))
-    # enter a valid zip and make sure we can continue
-    fill_in t('in_person_proofing.form.state_id.zipcode'), with: '123456789'
-    expect(page).to have_field(t('in_person_proofing.form.state_id.zipcode'), with: '12345-6789')
-    click_idv_continue
-    expect(page).to have_current_path(idv_in_person_ssn_url)
+  context 'when visiting state id for the first time' do
+    it 'displays correct heading and button text', allow_browser_log: true do
+      complete_steps_up_to_location_step
+
+      expect(page).to have_content(t('forms.buttons.continue'))
+      expect(page).to have_content(
+        t(
+          'in_person_proofing.headings.state_id_milestone_2',
+        ).tr(' ', ' '),
+      )
+    end
+
+    it 'allows the user to cancel and start over', allow_browser_log: true do
+      complete_steps_up_to_location_step
+
+      expect(page).not_to have_content('forms.buttons.back')
+
+      click_link t('links.cancel')
+      click_on t('idv.cancel.actions.start_over')
+      expect(page).to have_current_path(idv_welcome_path)
+    end
+
+    it 'allows the user to cancel and return', allow_browser_log: true do
+      complete_steps_up_to_location_step
+
+      expect(page).not_to have_content('forms.buttons.back')
+
+      click_link t('links.cancel')
+      click_on t('idv.cancel.actions.keep_going')
+      expect(page).to have_current_path(idv_in_person_step_path(step: :state_id), wait: 10)
+    end
+
+    it 'allows user to submit valid inputs on form', allow_browser_log: true do
+      complete_steps_up_to_location_step
+      fill_out_state_id_form_ok(same_address_as_id: true)
+      click_idv_continue
+
+      expect(page).to have_current_path(idv_in_person_ssn_url, wait: 10)
+      complete_ssn_step
+
+      expect_in_person_step_indicator_current_step(t('step_indicator.flows.idv.verify_info'))
+      expect(page).to have_current_path(idv_in_person_verify_info_url)
+      expect(page).to have_text(InPersonHelper::GOOD_FIRST_NAME)
+      expect(page).to have_text(InPersonHelper::GOOD_LAST_NAME)
+      expect(page).to have_text(InPersonHelper::GOOD_DOB_FORMATTED_EVENT)
+      expect(page).to have_text(InPersonHelper::GOOD_STATE_ID_NUMBER)
+      expect(page).to have_text(InPersonHelper::GOOD_IDENTITY_DOC_ADDRESS1)
+      expect(page).to have_text(InPersonHelper::GOOD_IDENTITY_DOC_ADDRESS2)
+      expect(page).to have_text(InPersonHelper::GOOD_IDENTITY_DOC_CITY)
+      expect(page).to have_text(InPersonHelper::GOOD_IDENTITY_DOC_ZIPCODE)
+    end
+  end
+
+  context 'when updating by visiting state id from verify info pg' do
```

Comment:
thoughts?

Reply:
Is there a reasonable expectation that within the context of this test, the result may be different (due to variables) than the verify_info spec?

Reply:
no, nothing should be different.